runtime/src/main/java/org/corfudb/runtime/Restore.java

```diff
@@ -57,11 +65,12 @@
      * @param runtime       - the runtime which is performing the restore
      * @throws IOException when failed to create the temp directory
      */
-    public Restore(String filePath, CorfuRuntime runtime) throws IOException {
+    public Restore(String filePath, CorfuRuntime runtime, boolean clearAllTablesBeforeRestore) throws IOException {
```

Comment:
would be great to have an enum instead of boolean, like:
`enum PostRestore{
  CLEAN, EMPTY_ACTION
}
`

Reply:
This flag is used to indicate if we clean up tables BEFORE restore. I added a enum `RestoreMode` and it has `FULL` (clean up all tables) and `PARTIAL` (only clean up tables that needs to be restored). This makes the API easier to understand.